### PR TITLE
tests: add test details for services-related spread tests, and a few others

### PR DESCRIPTION
**Pull request by olivercalder**

### CODE REVIEW

File: tests/main/snap-set/task.yaml [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+    2. If a configure hook errors, `snap set` fails.
+
+    3. Configuration options set via `snap set` can be retrieved by the
+    resulting configure hook.
+
+    4. Configuration options can be set to `null`, which removes the option.
+
+    5. Configuration options can be unset, which removes the option.
+
+    6. Setting an invalid key results in an error.
```
Comment (olivercalder) [inline tests/main/snap-set/task.yaml:19]: It is unclear to me where the valid keys are defined. In the the test `snapctl-hooks` snap, for example, the only thing special there seems to be about the `command` key is that it is checked as follows:
```
command=$(snapctl get command)
case $command in
    ...
```
Is the distinction that there must be a variable with a name matching the key defined at some point in the configure hook?

File: tests/lib/snaps/test-snapd-service/meta/hooks/configure [hunk 2]
```diff
@@ -4,11 +4,7 @@ snapctl set service-option="$(snapctl get service-option-source)"
 
 COMMAND=$(snapctl get command)
 if [ "$COMMAND" != "" ]; then
-    if [ "$COMMAND" = "restart" ]; then
```
Comment (sergiocazzolato) [inline tests/lib/snaps/test-snapd-service/meta/hooks/configure:7]: is this change required? is this fixing/simplifying the test?

File: tests/main/security-devpts/task.yaml [hunk 3]
```diff
@@ -1,5 +1,10 @@
 summary: Ensure that the basic devpts security rules are in place.
 
+details: |
+    Test that a snap which has a plug for the physical-memory-observe interface
+    can read /dev/pts, can openpty, and can access the slave PTY via /dev/ptmx,
```
Comment (sergiocazzolato) [inline tests/main/security-devpts/task.yaml:5]: we need to avoid the 'slave' word

File: tests/main/services-disable-install-hook/task.yaml [hunk 4]
```diff
@@ -1,5 +1,10 @@
 summary: Check that `snapctl stop --disable` actually stops services on install
 
+details: |
+    Install a snap with both simple and forking services and an install hook
+    which include `snapctl stop --disable`, then check that the install hook
```
Comment (andrewphelpsj) [inline tests/main/services-disable-install-hook/task.yaml:5]:
```suggestion
    which includes `snapctl stop --disable`, then check that the install hook
```

File: tests/lib/snaps/test-snapd-service/meta/hooks/configure (hunk 2 above)
Comment (olivercalder) [reply tests/lib/snaps/test-snapd-service/meta/hooks/configure:7]: It's not required, but it seems the `if` and `else` result in entirely identical behavior, so not sure why they're differentiated.

File: tests/main/security-devpts/task.yaml (hunk 3 above)
Comment (olivercalder) [reply tests/main/security-devpts/task.yaml:5]: I thought this was the (unfortunate) technical term, found in `man pty` and `man openpty`. Do you have a suggestion for what to call it instead?

File: tests/lib/snaps/test-snapd-service/meta/hooks/configure (hunk 2 above)
Comment (andrewphelpsj) [reply tests/lib/snaps/test-snapd-service/meta/hooks/configure:7]: In my opinion probably best to avoid actual changes here since this'll probably end up with the "Skip spread" tag, although this does look correct and simpler.

File: tests/main/security-devpts/task.yaml (hunk 3 above)
Comment (olivercalder) [reply tests/main/security-devpts/task.yaml:5]: Perhaps I can just `s/slave //`, since someone who understands PTY knows that it's the slave PTY which is found at `/dev/ptmx`

**Review by andrewphelpsj — APPROVED** [review]
Looks good, once you fix that linting error

**Review by sergiocazzolato — APPROVED** [review]
